src/main/java/com/icthh/xm/uaa/security/CachePasswordEncoder.java

```diff
@@ -0,0 +1,42 @@
+package com.icthh.xm.uaa.security;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+public class CachePasswordEncoder implements PasswordEncoder {
+
+        private final PasswordEncoder passwordEncoder;
+        private final Cache<String, CharSequence> passwordCache;
+
+        public CachePasswordEncoder(PasswordEncoder passwordEncoder, Integer cacheSize) {
+            this.passwordEncoder = passwordEncoder;
+            this.passwordCache = CacheBuilder.newBuilder()
+                .maximumSize(cacheSize != null ? cacheSize : 1000)
```

Comment:
pls consider moving parameter to config

Reply:
sorry, it is already in the config :). 
I've missed it because it is a constant number. So just move it to constant like DEFAULT_CAHCE_SIZE